drivers/i2c/i2c_mcux_flexcomm.c

```diff
@@ -59,7 +62,16 @@ static int mcux_flexcomm_configure(const struct device *dev,
 		return -EINVAL;
 	}
 
-	clock_freq = MHZ(12);
+	/* Get the clock frequency */
+	clock_dev = device_get_binding(config->clock_name);
```

Comment:
maybe it would be better to keep that pointer into device's private data. device_get_binding() is greedy, performance wise. So run it only once at init as you do already and keep the pointer.

Reply:
I have added this change